lbrynet/lbrynet_daemon/Daemon.py

```diff
@@ -660,17 +660,15 @@ def _download_name(self, name, stream_info, claim_id, timeout=None, download_dir
             self.analytics_manager.send_download_started(download_id, name, stream_info)
 
             self.streams[claim_id] = GetStream(self.sd_identifier, self.session,
-                                               self.session.wallet, self.lbry_file_manager,
                                                self.exchange_rate_manager, self.max_key_fee,
                                                conf.settings['data_rate'], timeout,
                                                download_directory, file_name)
             try:
-                download = self.streams[claim_id].start(stream_info, name)
+                lbry_file = yield self.streams[claim_id].start(stream_info, name)
```

Comment:
this will block adding the download-finished analytics event callback until after `start` returns, and the callback can't be added if finished_deferred is `None`.